src/crypto-api/index.ts

```diff
@@ -273,7 +273,8 @@ export interface CryptoApi {
     isCrossSigningReady(): Promise<boolean>;
 
     /**
-     * Get the ID of one of the user's cross-signing keys.
+     * Get the ID of one of the user's cross-signing keys, if the private is available.
+     * If the private key is not available, this will return null.
```

Comment:
missing "key", but it might be good to be more explicit about what "is available" means. Does "in 4S" count as "available"? (I think not?)

This is already touched on in the `@returns` tag, but I don't think "cross-signing has been initialised" is very clear either. It would be good to clarify that while you're here.

Reply:
Good question, looks like it depends what `hasMaster` etc. mean on `olmMachine.crossSigningStatus`. Do you have any idea without me delving into the rust?

Reply:
https://matrix-org.github.io/matrix-rust-sdk/matrix_sdk_crypto/olm/struct.CrossSigningStatus.html might be vaguely helpful.

It means "cached in the local crypto store"; ie the private key has been loaded from 4S or similar.

I also note that this will return null if we have the private key but not the public key (which can happen if we haven't yet done a /keys/query, or if the device that created the keys failed to publish them)

Reply:
Thanks - hopefully better now.